### ISSUE TITLE
Does LightGBM use Newton-Raphson optimization or not? 

### ISSUE THREAD

**Issue by nd7141** [microsoft/LightGBM#5233]
From the [LightGBM paper](https://proceedings.neurips.cc/paper/2017/file/6449f44a102fde848669bdd9eb6b76fa-Paper.pdf) I can see that each new tree approximates minus gradient. 

![image](https://user-images.githubusercontent.com/4607709/169873767-2f6cf91a-3464-4a3f-9a41-abbcaac21766.png)

Is it true? Or it approximates -gradient/hessian? 

The reason why I ask is that [XGBoost](https://arxiv.org/abs/1603.02754) seems to optimize Newton-Raphson update, which is -gradient/hessian. So I thought for LightGBM. 

In general I would really appreciate if you can help me to understand 3 steps of building GBDT model in LightGBM: 

1. What target variable each tree approximates? -gradient or -gradient/hessian? 
2. Given (X, y) how does LightGBM build a tree? Does it select the split with the highest loss reduction?  
3. How does it assign the leaf values? is it a simple average of the target variables that appear in the leaf?

**Comment by guolinke** [comment]
It is -grad/hess.
It choices the leaf with the max loss reduction.
It is -sum_grad/sum_hess.

In paper, the algorithm is grad only for simplicity.
